editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5717,46 +5719,57 @@ void CanvasItemEditorViewport::_on_change_type_confirmed() {
 	CheckBox *check = Object::cast_to<CheckBox>(button_group->get_pressed_button());
 	default_texture_node_type = check->get_text();
 	_perform_drop_data();
-	selector->hide();
+	texture_node_type_selector->hide();
 }
 
 void CanvasItemEditorViewport::_on_change_type_closed() {
 	_remove_preview();
 }
 
 void CanvasItemEditorViewport::_create_preview(const Vector<String> &files) const {
+	const String desc = TTR("Drag and drop to add as sibling of selected node (except when root is selected).") +
+			"\n" + TTR("Hold Shift when dropping to add as child of selected node.") +
+			"\n" + TTR("Hold Alt when dropping to add as child of root node.");
+
 	bool add_preview = false;
 	for (int i = 0; i < files.size(); i++) {
 		Ref<Resource> res = ResourceLoader::load(files[i]);
-		ERR_FAIL_COND(res.is_null());
+		ERR_CONTINUE(res.is_null());
+
 		Ref<Texture2D> texture = Ref<Texture2D>(Object::cast_to<Texture2D>(*res));
+		if (texture.is_valid()) {
+			Sprite2D *sprite = memnew(Sprite2D);
+			sprite->set_texture(texture);
+			sprite->set_modulate(Color(1, 1, 1, 0.7f));
+			preview_node->add_child(sprite);
+			label->show();
+			label_desc->set_text(desc + "\n" + TTR("Hold Alt + Shift when dropping to add as a different node type."));
```

Comment:
This is a pre-existing issue, but since this runs in a for loop, the resulting text can be a combination of strings that depends on order of files. E.g. texture header without type mention:
![image](https://github.com/godotengine/godot/assets/2223172/04b2dc03-4c9d-4a30-afea-46c233d84103)